lib/merchant_repository.rb

```diff
@@ -0,0 +1,56 @@
+require 'CSV'
+# require './lib/merchant'
+
+class MerchantRepository
+  attr_reader :all_merchants
+
+  def initialize(path)
+    @all_merchants = create_merchants(path)
+  end
+
+  def create_merchants(path)
+  @merchants = CSV.foreach(merchants, headers: true, header_converters: :symbol).map do |merchant|
+    Merchant.new(merchant, self)
+    end
+  end
+
+  def find_by_id(id)
+    @all_merchants.find do |merchant|
+      merchant.id == id
+    end
+  end
+
+  def find_by_name(name)
+    @all_merchants.find do |merchant|
+      merchant.name.downcase == name.downcase
```

Comment:
@leehopper used upcase, we should decide on a universal choice for this. I also tend to go with downcase, but it's completely a 50/50 choice here. :-)

Reply:
Agreed.